avocado/utils/nfs.py

```diff
@@ -0,0 +1,349 @@
+"""
+Basic nfs support for Linux host. It can support the remote
+nfs mount and the local nfs set up and mount.
+"""
+import re
+import os
+import logging
+import commands
+
+from . import path
+from . import process
+from . import service
+
+from utils_conn import SSHConnection
+import utils_misc
+
+
+def nfs_exported():
+    """
+    Get the list for nfs file system already exported
+
+    :return: a list of nfs that is already exported in system
+    :rtype: a lit of nfs file system exported
+    """
+    exportfs = process.system_output("exportfs -v")
+    if not exportfs:
+        return {}
+
+    nfs_exported_dict = {}
+    for fs_info in re.findall(r"[/\w+]+.*?\(.*?\)", exportfs, re.S):
+        fs_info = fs_info.strip().split()
+        if len(fs_info) == 2:
+            nfs_src = fs_info[0]
+            access_ip = re.findall(r"(.*)\(", fs_info[1])[0]
+            if "world" in access_ip:
+                access_ip = "*"
+            nfs_tag = "%s_%s" % (nfs_src, access_ip)
+            permission = re.findall(r"\((.*)\)", fs_info[1])[0]
+            nfs_exported_dict[nfs_tag] = permission
+
+    return nfs_exported_dict
+
+
+class Exportfs(object):
+
+    """
+    Add or remove one entry to exported nfs file system.
```

Comment:
Well it won't add or remove one entry, but it allows handling of the export location. (my point is it doesn't just add or remove the entry, it creates an instance which can be repeatedly set/reset or queried for status). Anyway if this is the original description than it should stay as is and we can add a second commit to improve it.